crates/configuration/src/shared/node.rs

```diff
@@ -152,16 +168,30 @@ impl NodeConfig {
         self.image.as_ref()
     }
 
+    /// The default container image used for nodes.
+    pub fn set_image(&mut self, image: Image) {
```

Comment:
Be careful of what we expose as public API

```suggestion
    pub(crate) fn set_image(&mut self, image: Image) {
```